cloudsync/tasks.py

```diff
@@ -165,23 +223,51 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
             video.update_status(VideoStatus.TRANSCODING)
 
 
-@shared_task(bind=True)
-def update_video_statuses(self):  # pylint: disable=unused-argument
+def process_watch_file(key):
     """
-    Check on statuses of all transcoding videos and update their status if appropriate
+    Copy the file from the watch bucket to the upload bucket, create model objects, and transcode
+
+    Args:
+        key(str): S3 object key
+
     """
-    transcoding_videos = Video.objects.filter(status=VideoStatus.TRANSCODING)
-    for video in transcoding_videos:
-        try:
-            refresh_status(video)
-        except EncodeJob.DoesNotExist:
-            # Log the exception but don't raise it so other videos can be checked.
-            log.exception("No EncodeJob object exists for video id %d", video.id)
-            video.update_status(VideoStatus.TRANSCODE_FAILED)
-        except ClientError as exc:
-            # Log the exception but don't raise it so other videos can be checked.
-            log.exception("AWS error when refreshing job status for video %d: %s", video.id, exc.response)
-            video.update_status(VideoStatus.TRANSCODE_FAILED)
+    watch_bucket = get_bucket(settings.VIDEO_S3_WATCH_BUCKET)
+
+    title = extract_title(key)
+    collection, _ = Collection.objects.get_or_create(title=title,
+                                                     owner=User.objects.get(username=settings.LECTURE_CAPTURE_USER))
+
+    # Create the necessary models
+    with transaction.atomic():
+        video = Video.objects.create(
+            source_url='https://{}/{}/{}'.format(settings.AWS_S3_DOMAIN, settings.VIDEO_S3_WATCH_BUCKET, key),
+            collection=collection,
+            title=key,
+            multiangle=True  # Assume all videos in watch bucket are multi-angle
+        )
+        video_file = VideoFile.objects.create(
+            s3_object_key=video.get_s3_key(),
+            video_id=video.id,
+            bucket_name=settings.VIDEO_S3_BUCKET
+        )
+
+    # Copy the file to the upload bucket using a new s3 key
+    s3_client = get_s3_client()
+    copy_source = {
+        'Bucket': watch_bucket.name,
+        'Key': key
+    }
+    try:
+        s3_client.copy(copy_source, settings.VIDEO_S3_BUCKET, video_file.s3_object_key)
+    except:
+        video.delete()
+        raise
+
+    # Delete the original file from the watch bucket
+    s3_client.delete_object(Bucket=settings.VIDEO_S3_WATCH_BUCKET, Key=key)
```

Comment:
If it fails then it would likely be imported again on the next task run, leading to a dupe video in the collection.  In that case maybe it would be best to delete the video & videofile just created?